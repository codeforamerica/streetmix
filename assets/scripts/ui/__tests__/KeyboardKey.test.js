/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import KeyboardKey from '../KeyboardKey'

describe('KeyboardKey', () => {
  it('renders a <kbd> element with string child', () => {
    const wrapper = shallow(<KeyboardKey>foo</KeyboardKey>)
    expect(wrapper.props().title).toBeUndefined()
    expect(wrapper.debug()).toMatchSnapshot()
  })

  it('renders a <kbd> element with element child', () => {
    const wrapper = shallow(<KeyboardKey><strong>foo</strong></KeyboardKey>)
    expect(wrapper.props().title).toBeUndefined()
    expect(wrapper.debug()).toMatchSnapshot()
  })

  it('renders a <kbd> element with icon and title', () => {
    const wrapper = shallow(<KeyboardKey icon={{ prefix: 'fas', iconName: 'minus' }}>foo</KeyboardKey>)
    expect(wrapper.find('FontAwesomeIcon').length).toBe(1)
    expect(wrapper.props().title).toBe('foo')
    expect(wrapper.debug()).toMatchSnapshot()
  })
})
