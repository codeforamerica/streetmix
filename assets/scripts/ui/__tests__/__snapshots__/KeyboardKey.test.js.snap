// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardKey renders a <kbd> element with element child 1`] = `
"<kbd className=\\"key\\">
  <strong>
    foo
  </strong>
</kbd>"
`;

exports[`KeyboardKey renders a <kbd> element with icon and title 1`] = `
"<kbd className=\\"key key-icon\\" title=\\"foo\\">
  <FontAwesomeIcon icon={{...}} />
</kbd>"
`;

exports[`KeyboardKey renders a <kbd> element with string child 1`] = `
"<kbd className=\\"key\\">
  foo
</kbd>"
`;
